giftcard/src/main/java/com/adyen/checkout/giftcard/SessionsGiftCardComponentCallback.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2023 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by ozgur on 28/3/2023.
+ */
+
+package com.adyen.checkout.giftcard
+
+import androidx.annotation.RestrictTo
+import com.adyen.checkout.components.core.BalanceResult
+import com.adyen.checkout.components.core.OrderResponse
+import com.adyen.checkout.components.core.PaymentComponentData
+import com.adyen.checkout.components.core.paymentmethod.GiftCardPaymentMethod
+import com.adyen.checkout.components.core.paymentmethod.PaymentMethodDetails
+import com.adyen.checkout.core.exception.MethodNotImplementedException
+import com.adyen.checkout.sessions.core.SessionComponentCallback
+import com.adyen.checkout.sessions.core.SessionPaymentResult
+import org.json.JSONObject
+
+/**
+ * Implement this callback to interact with a [GiftCardComponent] initialized with a session.
+ */
+interface SessionsGiftCardComponentCallback : SessionComponentCallback<GiftCardComponentState> {
+
+    /**
+     * Indicates that a partial payment has been done. This means an order for this payment has been created and
+     * part of the amount is still remaining to be paid. This callback provides you with the necessary objects to
+     * be able to create a new session to complete the payment for the remaining amount.
+     *
+     * @param result The result of the payment.
+     */
+    fun onPartialPayment(result: SessionPaymentResult)
+
+    // API Events
+    /**
+     * Only applicable for partial payments flow.
+     *
+     * Override this method if you want to take over the sessions flow and make a network call to the
+     * /paymentMethods/balance endpoint of the Checkout API through your server. This method is called right after the
+     * user enters their partial payment method details and submits them.
+     *
+     * You need to return [true] if you want to take over the sessions flow, otherwise the API calls will still be
+     * handled internally by the SDK. This could be useful in case you want to handle the flow yourself only in certain
+     * conditions, then you can return [false] if these conditions are not met.
+     *
+     * Once you take over the flow you will need to handle all the necessary subsequent network calls, otherwise a
+     * [MethodNotImplementedException] will be thrown.
+     *
+     * We provide a [PaymentMethodDetails] object that contains a non-serialized version of the partial payment method
```

Comment:
We should make sure to update these docs when we agree on the type of class we return